File: Frontend/src/events/converters/journals/quests/step.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { StepDTO } from "@/events/dto/journals/quests/step.dto";
import { RequisiteConverter } from "./requisite.converter";
import type { Step } from "@/models";

export class StepConverter {
  public static convert(stepDto: DeepRequired<StepDTO>): Step {
    return {
      number: stepDto.number,
      isDone: stepDto.isDone,
      requisites: stepDto.requisites.map((requisiteDto) => RequisiteConverter.convert(requisiteDto)),
    };
  }
}

File: Backend/Events/Converters/Journals/Quests/StepConverter.cs
using Backend.Domain.Models.Journals.Quests;
using Backend.Events.DTO.Journals.Quests;

namespace Backend.Events.Converters.Journals.Quests;

public static class StepConverter
{
    public static StepDTO ToDTO(Step step) => new()
    {
        Number = step.Number,
        IsDone = step.IsDone,
        Requisites = step.Requisites.Select(RequisiteConverter.ToDTO).ToList()
    };
}
